src/client/gfs_client.h

```diff
@@ -22,19 +22,27 @@ struct Data {
   void* buffer;
 };
 
+// Initialization function that a user should call before using any of the 
+// APIs below. This function takes a configuration file path, a master node
+// name, and an optional bool to indicate whether to resolve host name. 
+// It will initialize the internal units (e.g. cacheManager) according to the 
+// configuration.  
+google::protobuf::util::Status init_client(const std::string& config_filename,
+    const std::string& master_name, const bool resolve_hostname = false);
+
 // We support the following mode: Read Mode | Write Mode | Create Mode
 // when opening a file. The only possible combination is Write | Create  
-google::protobuf::util::Status open(const char* pathname, unsigned int flags);
+google::protobuf::util::Status open(const char* filename, unsigned int flags);
```

Comment:
For easy readability, please add a comment/documentation in listing the int values for each of the four modes, so caller can quickly understand the int->flag mapping as well

Reply:
Sounds good. Added.